databroker/core.py

```diff
@@ -944,7 +944,16 @@ def read_canonical(self):
             "may be removed in a future release.")
         yield from self.canonical(fill='yes')
 
+
     def __repr__(self):
+        try:
+            self._load()
+            start = self.metadata['start']
+            return f"<{self.__class__.__name__} uid={start['uid']!r}>"
+        except Exception as exc:
+            return f"<{self.__class__.__name__} *REPR RENDERING FAILURE* {exc!r}>"
+
+    def _repr_pretty_(self, p, cycle):
         self._load()
```

Comment:
Maybe this load should be in the try block. The same way that it is done in __repr__.

Reply:
Good catch, thanks.